docs/KIALI_INTEGRATION.md

```diff
@@ -0,0 +1,170 @@
+## Kiali integration
+
+This server can expose Kiali tools so assistants can query mesh information (e.g., mesh status/graph).
+
+### Enable the Kiali toolset
+
+Enable the Kiali tools via the server TOML configuration file.
+
+Config (TOML):
+
+```toml
+toolsets = ["core", "kiali"]
+
+[toolset_configs.kiali]
+url = "https://kiali.example"
+# insecure = true  # optional: allow insecure TLS (not recommended in production)
+# certificate_authority = """-----BEGIN CERTIFICATE-----
+# MIID...
+# -----END CERTIFICATE-----"""
+# When url is https and insecure is false, certificate_authority is required.
+```
+
+When the `kiali` toolset is enabled, a Kiali toolset configuration is required via `[toolset_configs.kiali]`. If missing or invalid, the server will refuse to start.
+
+### How authentication works
+
+- The server uses your existing Kubernetes credentials (from kubeconfig or in-cluster) to set a bearer token for Kiali calls.
+- If you pass an HTTP Authorization header to the MCP HTTP endpoint, that is not required for Kiali; Kiali calls use the server's configured token.
+
+### Available tools (initial)
+
+<details>
```

Comment:
This essentially duplicates what's shown in the main README.md file but it's not automatically generated.
As a follow-up we might want to either generate this automatically, or remove it from here and link to the main readme.